schemas/content/product-attribute.schema.json

```diff
@@ -0,0 +1,52 @@
+{
+  "meta:license": [
+    "Copyright 2018 Adobe Systems Incorporated. All rights reserved.",
+    "This work is licensed under a Creative Commons Attribution 4.0 International (CC BY 4.0) license",
+    "you may not use this file except in compliance with the License. You may obtain a copy",
+    "of the License at https://creativecommons.org/licenses/by/4.0/"
+  ],
+  "$id": "https://ns.adobe.com/xdm/content/product-attribute",
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "title": "Product Attribute",
+  "type": "object",
+  "meta:extensible": true,
+  "description": "Generic product attribute of a product in the product catalog.",
+  "definitions": {
+    "attribute": {
+      "properties": {
+        "xdm:key": {
+          "title": "Key",
+          "type": "string",
+          "description": "The technical key / identifier of this attribute."
+        },
+        "schema:name": {
+          "title": "Name",
+          "type": "string",
+          "description": "The human-readable name of the attribute."
+        },
+        "xdm:value": {
+          "title": "Value",
+          "description": "The attribute value."
+        },
+        "xdm:unit": {
+          "title": "Unit",
+          "type": "string",
+          "description": "The unit of measure for the attribute value."
+        },
+        "xdm:variantAttribute": {
+          "title": "Variant Attribute",
+          "type": "boolean",
+          "default": false,
+          "description": "Whether this attribute is a defining variant attribute which is used to distinguishe different product variants from each other."
```

Comment:
How does a boolean of true for being a variant "distinguishe different product variants from each other."

Typo: distinguish

Reply:
A true/false marker should be enough. For example for a shirt color & size attributes this is set to true, for attributes like brand or fabrication it is not.
This additional information for color & size give an indication the client that these attributes will influence the product variant. While brand for example has only a descriptive character.

Clients can for example display these as selections boxes and hence has influence on the "selected" SKU.